lib/internal/warn.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+// A process Warning is similar in structure to an Error except that they
+// are emitted by the process.on('warn') event rather than being thrown.
+
+const _message = Symbol('message');
+const _name = Symbol('name');
+const prefix = `(${process.release.name}:${process.pid}) `;
+
+class Warning {
+  constructor(name, message, constructor) {
+    this[_name] = name;
+    this[_message] = `${prefix}${message}`;
+    Error.captureStackTrace(this, constructor || Warning);
+  }
+
+  get message() {
+    return this[_message];
+  }
+
+  get name() {
+    return this[_name];
+  }
+}
+
+function emit(warning) {
+  process.nextTick(() => {
+    process.emit('warn', warning);
+  });
```

Comment:
Hmmm this seems fine but I don't know if I have the experience to validate that. Does anyone have specific problems that happen due to behavior like this and errors? (Well other than uncatchability, but we don't care about that here anyways.)